Runtime/Core/BezierMesh.cs

```diff
@@ -0,0 +1,185 @@
+#if USING_SPLINES
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine.Splines;
+
+namespace UnityEngine.ProBuilder
+{
+    [RequireComponent(typeof(ProBuilderMesh))]
+    [RequireComponent(typeof(SplineContainer))]
+    [ExecuteInEditMode]
+    sealed class BezierMesh : MonoBehaviour
```

Comment:
It would be nice to have this class as public so that a user can create a BezierMesh from script.
In that case (moving the class to public), the sealed makes sense.
To do that though, more documentation is needed on the class itself and on the different fields.